### ISSUE TITLE
IOS 15 collectionView canPerformAction is not called for delete: action

### ISSUE THREAD

**Issue by recepkucek** [MessageKit/MessageKit#1650]
I implemented the following delegate function in the sub class. It is called for cut,copy and paste actions but not for delete action.
Also canPerformAction is deprecated , Can this be the reason ? Because this was working at the previos iOS versions.

`
override func collectionView(_ collectionView: UICollectionView, canPerformAction action: Selector, forItemAt indexPath:IndexPath, withSender sender: Any?) -> Bool {
            
    let message=messages[indexPath.section];
    if action == NSSelectorFromString("delete:") {
        return true;
    } else {
        return super.collectionView(collectionView, canPerformAction: action, forItemAt: indexPath, withSender: sender)
    }
}

`

- MessageKit 3.7.0
- iOS 15.1
- Swift 5
- XCode 13.1

**Comment by iceboxi** [comment]
Apple delete UIMenuControl support in UICollectionView at iOS 14.

those delegate are wloudn't be call
```
- (BOOL)collectionView:(UICollectionView *)collectionView shouldShowMenuForItemAtIndexPath:(NSIndexPath *)indexPath API_DEPRECATED_WITH_REPLACEMENT("collectionView:contextMenuConfigurationForItemAtIndexPath:", ios(6.0, 13.0));
- (BOOL)collectionView:(UICollectionView *)collectionView canPerformAction:(SEL)action forItemAtIndexPath:(NSIndexPath *)indexPath withSender:(nullable id)sender API_DEPRECATED_WITH_REPLACEMENT("collectionView:contextMenuConfigurationForItemAtIndexPath:", ios(6.0, 13.0));
- (void)collectionView:(UICollectionView *)collectionView performAction:(SEL)action forItemAtIndexPath:(NSIndexPath *)indexPath withSender:(nullable id)sender API_DEPRECATED_WITH_REPLACEMENT("collectionView:contextMenuConfigurationForItemAtIndexPath:", ios(6.0, 13.0));
```

you can run the same code in iOS12, it's work.

**Comment by recepkucek** [comment]
I implemented like the following and it is working.

`@available(iOS 13.0, *)
extension MessagesViewController : UICollectionViewDelegate{
    func collectionView(_ collectionView: UICollectionView, contextMenuConfigurationForItemAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
        configureContextMenu(indexPath: indexPath,collectionView:collectionView);
    }
    
    func configureContextMenu(indexPath: IndexPath,collectionView:UICollectionView) -> UIContextMenuConfiguration{
        
        let message=messages[indexPath.section];
        
        let context = UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { (action) -> UIMenu? in
            let copy = UIAction(title: "Copy", image: UIImage(systemName: "square.on.square"), identifier: nil, discoverabilityTitle: nil, state: .off) { (_) in
                
            }
            let delete = UIAction(title: "Delete", image: UIImage(systemName: "trash"), identifier: nil, discoverabilityTitle: nil,attributes: .destructive, state: .off) { (_) in
                
            }
            let menu = UIMenu(title: "", image: nil, identifier: nil, options: UIMenu.Options.displayInline,
                              children: [copy,delete]);
            
            return menu;
        }
        return context
    }
}`

**Comment by wzio** [comment]
> I implemented like the following and it is working.

Hi, which version you used, I do same things like you, but it not work
